js/gdm/authPrompt.js

```diff
@@ -592,8 +619,14 @@ var AuthPrompt = class {
         // version had one less digit in the code).
         let resetCode = Main.customerSupport.passwordResetSalt ? '1' : '';
 
-        for (let n = 0; n < _RESET_CODE_LENGTH; n++)
-            resetCode = '%s%d'.format(resetCode, GLib.random_int_range(0, 10));
+        let machineId = this._getMachineId();
+        let lastLoginTime = this._getUserLastLoginTime();
+        let input = machineId + this._username + lastLoginTime;
+        let checksum = GLib.compute_checksum_for_data(GLib.ChecksumType.SHA256, input);
+        checksum = checksum.replace(/\D/g, '');
+
+        resetCode = '%s%d'.format(resetCode, parseInt(checksum) % (10 ** _RESET_CODE_LENGTH));
```

Comment:
Nitpick: can you use template strings here?

```javascript
        resetCode = `${resetCode}${parseInt(checksum) % (10 ** _RESET_CODE_LENGTH)}`;
```

Reply:
Done and actually fixed a small issue with padding the code to make sure it has the proper length.